tedana/resources/decision_trees/minimal_external1.json

```diff
@@ -0,0 +1,247 @@
+                "log_extra_info": "",
+                "log_extra_report": ""
+            },
+            "_comment": ""
+        },
+        {
+            "functionname": "calc_rho_elbow",
+            "parameters": {
+                "decide_comps": "all"
+            },
+            "kwargs": {
+                "subset_decide_comps": "unclassified",
+                "rho_elbow_type": "liberal",
+                "log_extra_info": "",
+                "log_extra_report": ""
+            },
+            "_comment": ""
+        },
+        {
+            "functionname": "dec_left_op_right",
+            "parameters": {
+                "if_true": "provisionalaccept",
+                "if_false": "provisionalreject",
+                "decide_comps": "unclassified",
+                "op": ">=",
+                "left": "kappa",
+                "right": "kappa_elbow_kundu"
+            },
+            "kwargs": {
+                "log_extra_report": ""
+            }
+        },
+        {
+            "functionname": "dec_left_op_right",
+            "parameters": {
+                "if_true": "accepted",
+                "if_false": "nochange",
+                "decide_comps": "provisionalaccept",
+                "op": ">",
+                "left": "kappa",
+                "right": "rho"
+            },
+            "kwargs": {
+                "log_extra_info": "If kappa>elbow and kappa>2*rho accept even if rho>elbow",
+                "log_extra_report": "",
+                "right_scale": 2,
+                "tag_if_true": "Likely BOLD"
+            }
+        },
+        {
+            "functionname": "dec_left_op_right",
+            "parameters": {
+                "if_true": "provisionalreject",
+                "if_false": "nochange",
+                "decide_comps": [
+                    "provisionalreject",
+                    "provisionalaccept"
+                ],
+                "op": ">",
+                "left": "rho",
+                "right": "rho_elbow_liberal"
+            },
+            "kwargs": {
+                "log_extra_report": ""
+            }
+        },
+        {
+            "functionname": "dec_variance_lessthan_thresholds",
+            "parameters": {
+                "if_true": "accepted",
+                "if_false": "nochange",
+                "decide_comps": "provisionalreject"
+            },
+            "kwargs": {
+                "var_metric": "variance explained",
+                "log_extra_info": "",
+                "log_extra_report": "",
+                "single_comp_threshold": 0.1,
+                "all_comp_threshold": 1.0,
+                "tag_if_true": "Low variance"
+            }
+        },
+        {
+            "functionname": "dec_left_op_right",
+            "parameters": {
+                "if_true": "rejected",
+                "if_false": "nochange",
+                "decide_comps": "all",
+                "op": ">",
+                "left": "^.*_correlation$",
```

Comment:
I think I need to write a function to expand trees if they contain regular expressions.

Reply:
I've got it working.

Reply:
I think regular expressions are not going to end up being particularly useful. That is, I don't think an effective test will be "Are any of these 12 correlation values>0.2?" but I might be wrong. If you've already got it working and you think it might be useful, I'm fine including it.

Reply:
I think it'll be important both for rejecting motion-correlated components (e.g., if we want to do what ICA-AROMA does) and for accepting signal-related components (e.g., if users want to provide their own signal regressors, which was part of the initial motivation for #1009, I believe). In the latter case, I don't think the number of signal regressors will be hardcodable (is that a word?) in decision trees, since it might depend on individual task conditions, so having something like `^task_.*_correlation$` might be the way to go.

Reply:
Thinking about this more, you're describing an inefficient F test. If the goal is that a user provides 6 regressor time series and the goal is to see if those 6 time series model a significant amount of variance of an ICA component time series, then you put all 6 in a linear model and calculate an F statistic for the full model. Calculating a correlation for each of the 6 separately means, if each of the 6 model 10% of the variance (with a 20% threshold) then it would be labeled non-significant while an F test would show that the combination of all 6 models 60% of the variance.

Reply:
I'm happy to extend the `external` module to include other metrics if we don't want correlations, but there are also correlation-based alternatives that users might want (e.g., the "maximum robust correlation" metric from ICA-AROMA, which runs correlations with individual motion parameters).

Reply:
This is making me lean more towards a json config file. That way we can modularly add options as there is interest without dealing with input parameter bloat. The other advantage is that we can include a config file that's paired with a decision tree so that, as long as users label their columns to match the config file, the regressors will be handled under-the-hood. If they want their own column labels, with the same decision tree, they could just copy the regressor file, change the column labels, and use one config for an entire study.